src/analyzer.ts

```diff
@@ -52,6 +55,14 @@ function isDependencyExternal(url: string) {
   return parseUrl(url).protocol !== null || url.startsWith('//');
 }
 
+/**
+ * Get a longer, single-line error message for logging and exeption-handling
+ * analysis Warning objects.
```

Comment:
Add a TODO to test analysis before optimization. Once done we can use WarningPrinter because you can pass it a writable stream for output.